File: PYTHON/PROGRAMMERS/12951.py
# JadenCase 문자열 만들기
# lv 2

def solution(s):
    answer = ''
    first = True
    for c in s:
        if c == " ":
            answer += c
            first = True
            continue

        if first:
            if c.isalpha() and c.lower():
                answer += c.upper()
            else:
                answer += c
            first = False
            
        else:
            if c.isupper():
                answer += c.lower()
            else:
                answer += c       
    return answer

print(solution("apple    Bannana="))

File: JAVASCRIPT/PROGRAMMERS/12951.js
// JadenCase 문자열 만들기
// lv 2
// 문자열, 구현

function solution(s) {
  let answer = '';
  const checkAlpha = /[a-zA-Z]/; //영어
  let word = '';
  s = s.toLowerCase();
  for (let i = 0; i < s.length; i++) {
    if (s[i] === ' ') {
      answer += s[i];
      word = '';
    } else {
      if (word === '' && checkAlpha.test(s[i])) {
        answer += s[i].toUpperCase();
      } else {
        answer += s[i];
      }
      word += s[i];
    }
  }

  return answer;
}
